Code/GraphMol/FileParsers/MultithreadedSDMolSupplier.cpp

```diff
@@ -0,0 +1,262 @@
+//
+//  Copyright (C) 2020 Shrey Aryan
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+#include "MultithreadedSDMolSupplier.h"
+
+#include "FileParserUtils.h"
+
+namespace RDKit {
+MultithreadedSDMolSupplier::MultithreadedSDMolSupplier(
+    const std::string &fileName, bool sanitize, bool removeHs,
+    bool strictParsing, unsigned int numWriterThreads, size_t sizeInputQueue,
+    size_t sizeOutputQueue) {
+  dp_inStream = openAndCheckStream(fileName);
+  initFromSettings(true, sanitize, removeHs, strictParsing, numWriterThreads,
+                   sizeInputQueue, sizeOutputQueue);
+  POSTCONDITION(dp_inStream, "bad instream");
+  startThreads();
+}
+
+MultithreadedSDMolSupplier::MultithreadedSDMolSupplier(
+    std::istream *inStream, bool takeOwnership, bool sanitize, bool removeHs,
+    bool strictParsing, unsigned int numWriterThreads, size_t sizeInputQueue,
+    size_t sizeOutputQueue) {
+  PRECONDITION(inStream, "bad stream");
```

Comment:
Here you check inStream using PRECONDITION.  In the SMILES equivalent, you use CHECK_INVARIANT.  Why the difference?

Reply:
I am not sure about the difference, but I tried to be consistent. In the SmilesMolSupplier implementation, CHECK_INVARIANT is used [here](https://github.com/rdkit/rdkit/blob/b558de22c6c2526e97a55dafa3fce45fbbeaee5c/Code/GraphMol/FileParsers/SmilesMolSupplier.cpp#L36) whereas in the SDMolSupplier implementation PRECONDITION is used [here](https://github.com/rdkit/rdkit/blob/4c1ea25fdafd37a38aba161f228435adbc45dbd2/Code/GraphMol/FileParsers/SDMolSupplier.cpp#L48)